source/common/tcp_proxy/upstream.cc

```diff
@@ -290,7 +290,9 @@ void Http2Upstream::setRequestEncoder(Http::RequestEncoder& request_encoder, boo
                           Http::Headers::get().ProtocolValues.Bytestream);
   }
 
-  config_.headerEvaluator().evaluateHeaders(*headers, downstream_info_);
+  config_.headerEvaluator().evaluateHeaders(
+      *headers, *Http::StaticEmptyHeaders::get().request_headers,
+      *Http::StaticEmptyHeaders::get().response_headers, downstream_info_);
```

Comment:
not sure should we also create a simple wrapper function for evaluator to eliminate this change?

Reply:
I cannot use `evaluateHeaders` wrappers defined  in `router`, because `headerEvaluator() `returns reference to `HeaderEvaluator` pure virtual class, not `HeaderEvaluatorImpl`. I suggest we leave it as is for now.